salesforce/bulkwrite.go

```diff
@@ -430,3 +399,100 @@ func (c *Connector) getIncompleteJobResults(jobInfo *GetJobInfoResult) *JobResul
 
 	return jobResult
 }
+
+func (c *Connector) BulkDelete(ctx context.Context, params BulkJobParams) (*BulkJobResult, error) {
+	jobBody := map[string]any{
+		"operation": "delete",
+		"object":    params.ObjectName,
+	}
+
+	return c.BulkJob(ctx, params, jobBody)
+}
+
+//nolint:funlen,cyclop
+func (c *Connector) BulkJob(
```

Comment:
rewritten from the previous `BulkWrite()` method because below can be reused for `BulkDelete` and any bulk job.